### PR TITLE
Add missing fallback for latex rendering

### LINKED ISSUE #13759: Latex Formula and Interpolation results in Error
I don't know if this problem was already solved, but using julia v.4.0.1+pre8 I can't use this call to print my optimization solution:

```
display(md"$x_{1}$ = $(getValue(x1))")
```

I get the following error:

```
LoadError: MethodError: `latexinline` has no method matching latexinline(::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Float64)
Closest candidates are:
  latexinline(::IO, !Matched::Base.Markdown.LaTeX)
  latexinline(::IO, !Matched::Base.Markdown.Code)
  latexinline(::IO, !Matched::Array{T,1})
  ...
while loading In[41], in expression starting on line 19

 in latex at markdown/render/latex.jl:50
 in latex at markdown/render/latex.jl:23
 in writemime at markdown/render/latex.jl:139
 in sprint at iostream.jl:206
 in display_dict at /Users/diego/.julia/v0.4/IJulia/src/execute_request.jl:42
 in display at /Users/diego/.julia/v0.4/IJulia/src/inline.jl:35
 in display at multimedia.jl:151
```

I try to escape the $$ from latex but achieved no success. I'm guessing that due to `$` has different semantic at the same line for render latex formula and interpolation it is resulting in compiling error. My polite and more simple solution for this could be implementing the concat method for `md` strings or `md` and `AbstractString`.

e.g.,

```
md"$x_{1}$ " * "$(getValue(x1))"
```

Best,

### PR DESCRIPTION
**Pull request by MichaelHatherly**
Fixes #13759.

cc @dmoliveira, this should fix the problem you encountered. If you could confirm that would be great, thanks.

### CODE REVIEW

**Comment by dmoliveira** [comment]
Thanks for the effort @MichaelHatherly! It's almost works. I need to force every type to be string, for example:
```
x = [1,2,3]
md"$x_{1}$ = $x"
```
will results in error, but...

```
x = [1,2,3]
md"$x_{1}$ = $(string(x))"
```
will not.

I don't know if is easy to fix or it isn't a good practice to convert everything to string before printing, but doing it the effect will become consistent as in print/println that can print different interpolations types than AbstractString.

I tried this
```
latexinline(io::IO, x) = writemime(io, MIME"text/plain"(), string(x))
```

but didn't work.

**Comment by MichaelHatherly** [comment]
> will results in error

What's the error you're getting there?

It looks like interpolated vectors are getting dispatched to https://github.com/JuliaLang/julia/blob/5d7c0ade666e3051852ee9d50b33bf524e167e28/base/markdown/render/latex.jl#L78 when rendered. For you example I'm getting:

```
julia> x = [1,2,3];

julia> md"$x_{1}$ = $x"
x_{1}

  = 123
```

and might need some larger changes to make it work correctly.

> but didn't work.

Did it produce an error, or just not seem to have an effect?

**Comment by dmoliveira** [comment]
Yes, you are right @MichaelHatherly. It works on Julia REPL. 
But, please try the same code at Jupyter Notebook. Some errors happen.
The same example at Jupyter results in this:
```
MethodError: `latexinline` has no method matching latexinline(::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Int64)
Closest candidates are:
  latexinline(::IO, !Matched::Base.Markdown.LaTeX)
  latexinline(::IO, !Matched::Base.Markdown.Code)
  latexinline(::IO, !Matched::Array{T,1})
  ...

 in latexinline at markdown/render/latex.jl:80
 in latex at markdown/render/latex.jl:50
 in latex at markdown/render/latex.jl:23
 in writemime at markdown/render/latex.jl:139
 in sprint at iostream.jl:206
 in display_dict at /Users/diego/.julia/v0.4/IJulia/src/execute_request.jl:42
```

File: base/markdown/render/latex.jl [hunk 1]
```diff
@@ -136,5 +136,7 @@ latex(md) = sprint(latex, md)
 latexinline(md) = sprint(latexinline, md)
 latexesc(s) = sprint(latexesc, s)
 
+latexinline(io::IO, x) = writemime(io, MIME"text/plain"(), x)
```
Comment (hayd) [inline base/markdown/render/latex.jl]: I think this should `latexesc`?

I have a feeling I intentionally removed this...

*The idea being that users/libraries should be defining there own latex rendering of their type rather fallthough to string/markdown, and it should raise if they haven't.*

Comment (MichaelHatherly) [inline base/markdown/render/latex.jl]: > I think this should `latexesc`?

Yeah, should be. Thanks.

> *The idea being that users/libraries should be defining there own latex rendering of their type rather fallthough to string/markdown, and it should raise if they haven't.*

Is that written down somewhere? We could raise a slightly more informative error message if not.

**Comment by MichaelHatherly** [comment]
> But, please try the same code at Jupyter Notebook. Some errors happen.

2015-10-26-074009_4966x1080_scrot

It seems to work fine there as well. Is your notebook using the correct Julia kernel @dmoliveira?

**Comment by dmoliveira** [comment]
@MichaelHatherly maybe my Julia version is outdated.. I'm using `Julia 0.4.1-pre`:
latex_error_md

I cannot try now, but I'll update my code with master and come with a feedback soon. I think you can close the issue. Probably it is a local error after all. If it not works for me after that, than we can reopen the issue. Many thanks @MichaelHatherly and @hayd.

**Comment by kshyatt** [comment]
@dmoliveira did you get a chance to confirm whether it was just a local error? Is this ready to merge?

**Comment by hayd** [comment]
This should latexesc as discussed above.

I still kinda think that this should error, and a proper html/latex/etc output should be defined for array types (and other types we want to render). Specifically the fall-through output of html sucks/is useless:
```
julia> html(md"$x")
"1\n2\n3\n\n"
```

**Comment by MichaelHatherly** [comment]
Coming back to this (sorry for the delay), I think I agree with @hayd that some proper output methods should be defined rather than just a catch-all one like this.

Given the comment in https://github.com/JuliaLang/julia/issues/8963#issuecomment-153282305 about removing interpolation from `@doc_str`, presumably we won't need to define any of the non-markdown type rendering methods at all since there won't be any embedded values in the AST.

**Comment by tkelman** [comment]
still relevant?

**Comment by hayd** [comment]
Yes.

As more libs move to markdown/Documenter this may come up (when they start generating docs). Both latex and [html](https://github.com/JuliaLang/julia/pull/13760#issuecomment-151033732) fallthroughs appear to be useless atm.

**Comment by MichaelHatherly** [comment]
I'd prefer to just remove interpolation in `@doc_str`, not found it of any use personally.

**Pull request closed without merging**